rllib/core/models/tests/test_catalog.py

```diff
@@ -0,0 +1,150 @@
+            {
+                "fcnet_activation": "relu",
+                "fcnet_hiddens": [256, 256, 256],
+            },
+            {
+                "fcnet_hiddens": [512],
+                "fcnet_activation": "relu",
+            },
+            # {"use_lstm": True, "lstm_cell_size": 256},
+            # {"use_attention": True, "attention_num_transformer_units": 4},
+            # {
+            #     "fcnet_hiddens": [32],
+            #     "fcnet_activation": "linear",
+            #     "vf_share_layers": True,
+            #     "use_attention": True,
+            #     "max_seq_len": 10,
+            #     "attention_num_transformer_units": 1,
+            #     "attention_dim": 32,
+            #     "attention_memory_inference": 10,
+            #     "attention_memory_training": 10,
+            #     "attention_num_heads": 1,
+            #     "attention_head_dim": 32,
+            #     "attention_position_wise_mlp_dim": 32,
+            # },
+            # {
+            #     "use_lstm": True,
+            #     "conv_activation": "elu",
+            #     "dim": 42,
+            #     "grayscale": True,
+            #     "zero_mean": False,
+            #     # Reduced channel depth and kernel size from default
+            #     "conv_filters": [
+            #         [32, [3, 3], 2],
+            #         [32, [3, 3], 2],
+            #         [32, [3, 3], 2],
+            #         [32, [3, 3], 2],
+            #     ]
+            # }
+        ]
+
+        frameworks = ["tf", "torch"]
+
+        # First check if encoders can be created for singular spaces
+        print("Testing encoders for singular input spaces...")
+        config_combinations = [frameworks, input_spaces_and_config_types, model_configs]
+        for config in itertools.product(*config_combinations):
+            framework, input_space_and_config_type, model_config_dict = config
+            input_space, model_config_type = input_space_and_config_type
+            if model_config_type is not MLPEncoderConfig and framework == "tf":
+                # TODO (Artur): Enable this once we have TF implementations
+                continue
+            print(
+                f"Testing framework: \n{framework}\n, input space: \n{input_space}\n "
+                f"and config: \n{model_config_dict}\n"
+            )
+            catalog = Catalog(
+                observation_space=input_space,
+                # Action space does not matter for encoders
+                action_space=gym.spaces.Box(1, 1, (1,)),
+                model_config=model_config_dict,
+                # TODO(Artur): Add view requirements when we need them
+                view_requirements=None,
+            )
+
+            model_config = catalog.get_encoder_config(
+                observation_space=input_space, model_config=model_config_dict
+            )
+            assert type(model_config) == model_config_type
+            model_config.build(framework=framework)
```

Comment:
I think the next thing that should be tested in this loop is if you construct the encoder and feed a batch of sampled obs from the space into the encoder, you'll get the expected encoded features. 

```
encoder = model_config.build(fw)
sample_batch = create_batch(fw, batch_size=16, obs_space)
enc_out = encoder(sample_batch)
check_encoder_output(enc_out)
```

Reply:
You mean the dimensions should fit?

Reply:
❓

Reply:
@kouroshHakha I've added the test code you requested.

Reply:
yep that is what i meant.